### PR TITLE
[util, gen_cheader] Generate module parameters

### PR DESCRIPTION
**Pull request by silvestrst**
Until now some of the properties that are present in corresponding IP `hjson` files, were not auto-generated. And as a result had to be hardcoded in some cases.

Partially addresses #1710

### CODE REVIEW

File: util/reggen/gen_cheader.py [hunk 1]
```diff
@@ -149,8 +150,43 @@ def gen_cdefine_window(outstr, win, comp, regwidth, rnames, existing_defines):
...
+
+    # Presently there is only one type (int), however if the new types are
+    # added, they potentially need to be handled differently.
+    known_types = ["int"]
+    if param_type not in known_types:
+        warnings.warn("Cannot generate a module define")
+        return
+
+    genout(outstr, format_comment(first_line(param['desc'])))
+    define_name = as_define(module_name + '_' + param['name'])
```
[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: I don't think it's likely, but nevertheless I will make this change, it's a fair point.

File: util/reggen/gen_cheader.py [hunk 2]
```diff
@@ -149,8 +150,43 @@ def gen_cdefine_window(outstr, win, comp, regwidth, rnames, existing_defines):
...
+    module_params = set()
 
+    if 'param_list' in module_data:
+        module_params = module_data['param_list']
+
+    for param in module_params:
+        gen_cdefines_module_param(outstr, param, module_name, existing_defines)
+
+    genout(outstr, format_comment(first_line("Register width")))
+    define_name = as_define(module_name + '_' + "REG_WIDTH")
```
[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Same as above.

File: util/reggen/gen_cheader.py [hunk 3]
```diff
@@ -149,8 +150,43 @@ def gen_cdefine_window(outstr, win, comp, regwidth, rnames, existing_defines):
...
 
-# Must have called validate, so should have no errors
+def gen_cdefines_module_param(outstr, param, module_name, existing_defines):
+    param_type = param['type']
+
+    # Presently there is only one type (int), however if the new types are
+    # added, they potentially need to be handled differently.
+    known_types = ["int"]
+    if param_type not in known_types:
+        warnings.warn("Cannot generate a module define")
```
[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Will, do, not sure if it allows for `printf` like syntax, but I can always jam it in with `+ param_type +`.

[the remark this answers is not in the document]
Comment (lenary) [reply util/reggen/gen_cheader.py]: `"Cannot generate a module define of type {}".format(param_type)` will do the formatting you want :)

[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Thanks, thats handy!

[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Done.

File: util/reggen/gen_cheader.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Done.

File: util/reggen/gen_cheader.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (silvestrst) [reply util/reggen/gen_cheader.py]: Done.

**Pull request merged**